src/throttle/amoc_throttle.erl

```diff
@@ -1,133 +1,149 @@
-%% It also accepts a special value of `0' which limits the number of parallel executions associated with `Name' to `Rate'.
--spec start(name(), rate(), interval(), pos_integer()) -> ok | {error, any()}.
-start(Name, Rate, Interval, NoOfProcesses)
-  when is_atom(Name), ?POS_INT(Rate), ?NONNEG_INT(Interval), ?POS_INT(NoOfProcesses) ->
-    amoc_throttle_controller:ensure_throttle_processes_started(Name, Rate, Interval, NoOfProcesses);
-start(_Name, _Rate, _Interval, _NoOfProcesses) ->
-    {error, invalid_throttle}.
+-spec start(name(), t() | rate()) -> {ok, started | already_started} | {error, any()}.
+start(Name, #{} = Config) ->
+    amoc_throttle_controller:ensure_throttle_processes_started(Name, Config);
+start(Name, Rate) ->
+    amoc_throttle_controller:ensure_throttle_processes_started(Name, #{rate => Rate}).
 
 %% @doc Pauses executions for the given `Name' as if `Rate' was set to `0'.
 %%
-%% Does not stop the scheduled rate changes.
+%% Does not stop the scheduled rate changes. `resume/1' undoes the pausing.
 -spec pause(name()) -> ok | {error, any()}.
 pause(Name) ->
     amoc_throttle_controller:pause(Name).
 
+%% @doc Unlocks executions for the given `Name' as if `Rate' was set to `infinity'.
+-spec unlock(name()) -> ok | {error, any()}.
+unlock(Name) ->
+    amoc_throttle_controller:unlock(Name).
+
 %% @doc Resumes the executions for the given `Name', to their original `Rate' and `Interval' values.
+%%
+%% It is the counterpart to the `pause/1' API, resuming the execution of what that mechanism paused.
 -spec resume(name()) -> ok | {error, any()}.
 resume(Name) ->
     amoc_throttle_controller:resume(Name).
 
-%% @doc Sets `Rate' and `Interval' for `Name' according to the given values.
-%%
-%% Can change whether Amoc throttle limits `Name' to parallel executions or to `Rate' per `Interval',
-%% according to the given `Interval' value.
--spec change_rate(name(), rate(), interval()) -> ok | {error, any()}.
-change_rate(Name, Rate, Interval) ->
-    amoc_throttle_controller:change_rate(Name, Rate, Interval).
+%% @doc Sets the throttle `Config' for `Name' according to the given values.
+-spec change_rate(name(), t() | rate()) -> ok | {error, any()}.
+change_rate(Name, #{} = Config) ->
+    amoc_throttle_controller:change_rate(Name, Config);
+change_rate(Name, Rate) when is_integer(Rate) ->
+    amoc_throttle_controller:change_rate(Name, #{rate => Rate}).
 
 %% @doc Allows to set a plan of gradual rate changes for a given `Name'.
 %%
-%% `Rate' will be changed from `FromRate' to `ToRate' in a series of consecutive steps.
-%% Note that `FromRate' does not need to be lower than `ToRate', rates can be changed downwards.
+%% The configuration will be changed in a series of consecutive steps.
+%% Rates can be changed upwards as well as downwards.
 %%
 %% The rate is calculated at each step in relation to the `RateInterval', which can also be `0'.
 %% There will be `NoOfSteps' steps, each taking `StepInterval' time in milliseconds.
 %%
```

Comment:
please remove references to non-existing parameters (RateInterval,  NoOfSteps, StepInterval) and use corresponding atom keys in the configuration map.